Tests/SourceKitLSPTests/RewriteSourceKitPlaceholdersTests.swift

```diff
@@ -0,0 +1,62 @@
+import SKTestSupport
+@testable import SourceKitLSP
+import XCTest
+
+final class RewriteSourceKitPlaceholdersTests : XCTestCase {
+  func testClientDoesNotSupportSnippets() {
+    let input = "foo(bar: \(placeholder: "T##Int##Int#"))"
+    let rewritten = rewriteSourceKitPlaceholders(in: input, clientSupportsSnippets: false)
+
+    XCTAssertEqual(rewritten, "foo(bar: )")
+  }
+
+  func testInputWithoutPlaceholders() {
+    let input = "foo()"
+    let rewritten = rewriteSourceKitPlaceholders(in: input, clientSupportsSnippets: true)
+
+    XCTAssertEqual(rewritten, "foo()")
+  }
+
+  func testPlaceholderWithType() {
+    let input = "foo(bar: \(placeholder: "T##bar##Int"))"
+    let rewritten = rewriteSourceKitPlaceholders(in: input, clientSupportsSnippets: true)
+
+    XCTAssertEqual(rewritten, "foo(bar: ${1:Int})")
+  }
+
+  func testMultiplePlaceholders() {
+    let input = "foo(bar: \(placeholder: "T##Int##Int"), baz: \(placeholder: "T##String##String"), quux: \(placeholder: "T##String##String"))"
+    let rewritten = rewriteSourceKitPlaceholders(in: input, clientSupportsSnippets: true)
+
+    XCTAssertEqual(rewritten, "foo(bar: ${1:Int}, baz: ${2:String}, quux: ${3:String})")
+  }
+
+  func testClosurePlaceholderReturnType() {
+    let input = "foo(bar: \(placeholder: "{ \(placeholder: "T##Int##Int") }"))"
+    let rewritten = rewriteSourceKitPlaceholders(in: input, clientSupportsSnippets: true)
+
+    XCTAssertEqual(rewritten, "foo(bar: ${1:\\{ ${2:Int} \\}})")
```

Comment:
You can use raw string literals so you don’t need to write double backslashes here. I think that would improve readability.

```suggestion
    XCTAssertEqual(rewritten, #"foo(bar: ${1:\{ ${2:Int} \}})"#)
```